apps/context.go

```diff
@@ -75,8 +76,12 @@ type ExpandedContext struct {
 	User *model.User `json:"user,omitempty"`
 }
 
-// Paths for OAuth endpoint that the App needs to set up remote (3rd party) OAuth
-const (
-	PathOAuthRedirect = "/oauth2/remote/redirect"
-	PathOAuthComplete = "/oauth2/remote/complete"
-)
+type OAuth2Context struct {
+	// Expanded with "oauth2_app". Config must be previously stored with
+	// mmclient.StoreOAuth2App
+	OAuth2App
+	RedirectURL string `json:"redirect_url,omitempty"`
+	CompleteURL string `json:"complete_url,omitempty"`
+
+	User interface{} `json:"user,omitempty"`
```

Comment:
What kind of `User` is this?

Reply:
A generic 3rd party "user" object, up to the App. In hello-oauth it's `oauth2.Token`